src/views/ChallengeLinkLanding/ChallengeLinkLanding.tsx

```diff
@@ -0,0 +1,232 @@
+import { ChallengeDetailsReviewPane } from "ChallengeDetailsReviewPane";
+
+type Challenge = socket_api.seekgraph_global.Challenge;
+
+// Users are intended to arrive here via a challenge-link URL - those point here.
+// This page will display a bouncing OGS logo until the challenge given in search param is loaded.
+// From there, the user can accept the game - going via login orn auto-registration if necessary.
+
+export function ChallengeLinkLanding(): JSX.Element {
+    const user = data.get("user");
+    const logged_in = !user.anonymous;
+
+    /* State */
+    const [linked_challenge, set_linked_challenge] = React.useState<Challenge>(null);
+    const [ask_to_login, set_ask_to_login] = React.useState<boolean>(false);
+
+    /* Actions */
+
+    const doAcceptance = (challenge: Challenge) => {
+        swal({
+            text: "Accepting game...",
+            type: "info",
+            showCancelButton: false,
+            showConfirmButton: false,
+            allowEscapeKey: false,
+        }).catch(swal.noop);
+
+        post("challenges/%%/accept", challenge.challenge_id, {})
+            .then(() => {
+                swal.close();
+                browserHistory.push(`/game/${challenge.game_id}`);
+            })
+            .catch((err) => {
+                swal.close();
+                errorAlerter(err);
+            });
+        data.set("pending_accepted_challenge", null);
+    };
+
+    const accept = () => {
+        if (logged_in) {
+            doAcceptance(linked_challenge);
+        } else {
+            set_ask_to_login(true);
+            // We need to save the challenge info in this way for when we come back after logging in.
+            // An alternative is passing this "state" via the URL hash or params, but believe me
+            // this does not work out well :)
+            data.set("pending_accepted_challenge", linked_challenge);
+            console.log("pended", linked_challenge);
+        }
+    };
+
+    const registerAndCommence = () => {
+        const new_username_root = pgettext(
+            "This needs to be a no-spaces honorific for a guest user",
+            "HonouredGuest",
+        );
+        // naively uniquify... 6 lsbs of epoch... good enough?
+        const new_username =
+            new_username_root.replace(/\s+/g, "") + Date.now().toString().slice(-6);
+
+        const initial_password = Date.now().toString(); // They will have to change this, so anything random & unique is OK
+
+        post("/api/v0/register", {
+            username: new_username,
+            password: initial_password,
+            email: "",
+            ebi: get_ebi(),
+        })
+            .then((config) => {
+                data.set(cached.config, config);
+
+                dynamic_help.showHelpSet("guest-password-help-set");
+
+                doAcceptance(linked_challenge);
+            })
+            // note - no handling at the moment for the hopefully madly-unlikely duplicate username,
```

Comment:
Man, he ran the maths!    I guess I have to believe you, though strictly speaking there are 10^6 days in the year _per language_ of "honoured" users :p , so maybe that reduces the probabilities a teeny bit 🤣 

Still, those numbers are not at all unrealistic, disappointingly.

Interestingly, we can increase the number of digits safely at any time to start a new set, or indeed put in a new algorithm at any time with full backwards compatibility (excluding any algorithm that also generates 6 digit numbers I guess)  :)

I'm going to leave this for now, because I'm not sure we're going to like this specific system in the long run, though it's the best I could come up with without going nuts.